src/library-authoring/author-library/LibraryAuthoringContainer.jsx

```diff
@@ -0,0 +1,303 @@
+import React, {
+  useEffect,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import { v4 as uuid4 } from 'uuid';
+import { connect } from 'react-redux';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import {
+  clearLibrary,
+  clearLibraryError,
+  clearLibrarySuccess,
+  commitLibraryChanges,
+  createBlock,
+  fetchBlocks,
+  fetchLibraryDetail,
+  revertLibraryChanges,
+  searchLibrary,
+} from './data';
+import {
+  BLOCK_FILTER_ORDER,
+  LIBRARY_TYPES,
+  libraryBlockShape,
+  libraryShape,
+  LOADING_STATUS,
+  fetchable,
+  paginated,
+} from '../common';
+import { LoadingPage } from '../../generic';
+import messages from './messages';
+import { blockStatesShape } from '../edit-block/data/shapes';
+import commonMessages from '../common/messages';
+import selectLibraryDetail from '../common/data/selectors';
+import LibraryAuthoringPage from './LibraryAuthoringPage';
+
+const deriveTypeOptions = (blockTypes, intl) => {
+  let typeOptions = blockTypes.map((typeSpec) => (
+    { value: typeSpec.block_type, label: typeSpec.display_name }
+  ));
+  typeOptions.push({ value: '^', label: intl.formatMessage(messages['library.detail.other_component']) });
+  typeOptions = typeOptions.filter((entry) => BLOCK_FILTER_ORDER.includes(entry.value));
+  typeOptions.sort((a, b) => {
+    const aOrder = BLOCK_FILTER_ORDER.indexOf(a.value);
+    const bOrder = BLOCK_FILTER_ORDER.indexOf(b.value);
+    if (aOrder === bOrder) {
+      // Should never happen, but could cause problems if it did and we didn't indicate they should be treated the same.
+      return 0;
+    }
+    if (BLOCK_FILTER_ORDER.indexOf(a.value) > BLOCK_FILTER_ORDER.indexOf(b.value)) {
+      return 1;
+    }
+    return -1;
+  });
+
+  typeOptions.unshift({ value: '', label: intl.formatMessage(messages['library.detail.all_types']) });
+  return typeOptions;
+};
+
+/**
+ * LibraryAuthoringPageContainerBase
+ *
+ * Container for the Library Authoring page.
+ * This is the main page for the authoring tool.
+ */
+export const LibraryAuthoringPageContainerBase = ({
+  intl, library, blockStates, blocks, ...props
+}) => {
+  const { libraryId } = props.match.params;
+  const [query, setQuery] = useState('');
+  const [type, setType] = useState('');
+  const [page, setPage] = useState(1);
+  const [sending, setSending] = useState(false);
+
+  const pageSize = +process.env.LIBRARY_AUTHORING_PAGINATION_PAGE_SIZE;
+  const lastPage = Math.ceil(blocks.value.count / pageSize) || 1;
+  const paginationParams = {
+    page,
+    page_size: pageSize,
+  };
+
+  // On mount.
+  useEffect(() => {
+    if (!library || (library && library.id !== libraryId)) {
```

Comment:
```suggestion
    if (library?.id !== libraryId)) {
```
Probably should be the same

Reply:
I think you lost `!library` from the condition

Reply:
I just think that `libraryId ` can not be `undefined` so that single condition will cover this case too.

Reply:
updated